diff -up vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Command/AbstractCommand.php.patch vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Command/AbstractCommand.php
--- vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Command/AbstractCommand.php.patch	2014-03-05 10:48:57.298092011 +0100
+++ vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Command/AbstractCommand.php	2014-03-05 11:33:46.031956078 +0100
@@ -97,7 +97,12 @@ abstract class AbstractCommand extends C
             mkdir($dir, 0755, true);
         }
 
-        $templates = $this->findFilesInFolder($dir . '/../views', 'twig');
+        $templates = array();
+        if ( $name === 'app' ) {
+            $templates = $this->findFilesInFolder($dir . '/../../', 'twig');
+        } else {
+            $templates = $this->findFilesInFolder($dir . '/../views', 'twig');
+        }
 
         $php  = "<?php\n";
         $twig = $this->getContainer()->get('twig');
diff -up vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Routing/GettextLocaleListener.php.patch vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Routing/GettextLocaleListener.php
--- vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Routing/GettextLocaleListener.php.patch	2014-03-05 11:45:19.434819905 +0100
+++ vendor/leaseweb/gettext-translation-bundle/Lsw/GettextTranslationBundle/Routing/GettextLocaleListener.php	2014-03-05 11:45:28.325023766 +0100
@@ -43,10 +43,6 @@ class GettextLocaleListener
    */
   public function onKernelRequest(GetResponseEvent $event)
   {
-      if (HttpKernelInterface::MASTER_REQUEST !== $event->getRequestType()) {
-          return;
-      }
-
       $currentLocale = $this->getCurrentLocale();
       $requestedLocale = $this->getRequestedLocale();
 
