diff -up vendor/solarium/solarium/library/Solarium/QueryType/Select/ResponseParser/Component/FacetSet.php.patch vendor/solarium/solarium/library/Solarium/QueryType/Select/ResponseParser/Component/FacetSet.php
--- vendor/solarium/solarium/library/Solarium/QueryType/Select/ResponseParser/Component/FacetSet.php.patch	2014-03-11 12:05:01.497565361 +0100
+++ vendor/solarium/solarium/library/Solarium/QueryType/Select/ResponseParser/Component/FacetSet.php	2014-03-11 12:05:52.546739217 +0100
@@ -102,6 +102,7 @@ class FacetSet extends ResponseParserAbs
 
         $facets = array();
         foreach ($facetSet->getFacets() as $key => $facet) {
+            $key = current(explode(' ', $key));
             switch ($facet->getType()) {
                 case QueryFacetSet::FACET_FIELD:
                     $result = $this->facetField($query, $facet, $data);
@@ -217,7 +218,7 @@ class FacetSet extends ResponseParserAbs
      */
     protected function facetRange($query, $facet, $data)
     {
-        $key = $facet->getKey();
+        $key = current(explode(' ', $facet->getKey()));
         if (!isset($data['facet_counts']['facet_ranges'][$key])) {
             return null;
         }
